### PR TITLE
PMM-11844: fix passing additional auth parameters, update mongo driver

### PR DESCRIPTION
**Pull request by ritbl**
[PMM-11844](https://jira.percona.com/browse/PMM-11844)

Related PR:
 - https://github.com/percona/mongodb_exporter/pull/640

Feature Build:
 - https://github.com/Percona-Lab/pmm-submodules/pull/3158

### CODE REVIEW

File: agent/utils/mongo_fix/mongo_fix.go [hunk 1]
```diff
@@ -36,7 +36,16 @@ func ClientOptionsForDSN(dsn string) (*options.ClientOptions, error) { //nolint:
 	username := parsedDsn.User.Username()
 	password, _ := parsedDsn.User.Password()
 	if username != "" || password != "" {
-		clientOptions = clientOptions.SetAuth(options.Credential{Username: username, Password: password})
+		clientOptions = clientOptions.SetAuth(
+			options.Credential{
+				AuthMechanism:           clientOptions.Auth.AuthMechanism,
```
Comment (ritbl) [inline agent/utils/mongo_fix/mongo_fix.go:41]: Additional auth parameters should be passed from original options. see line 27

**Review by BupycHuk — APPROVED** [review]

File: agent/utils/mongo_fix/mongo_fix.go [hunk 2]
```diff
@@ -36,7 +36,16 @@ func ClientOptionsForDSN(dsn string) (*options.ClientOptions, error) { //nolint:
...
+				AuthMechanismProperties: clientOptions.Auth.AuthMechanismProperties,
+				AuthSource:              clientOptions.Auth.AuthSource,
+				Username:                username,
+				Password:                password,
+				PasswordSet:             clientOptions.Auth.PasswordSet,
+			},
+		)
```
Comment (artemgavrilov) [inline agent/utils/mongo_fix/mongo_fix.go:48]: I'm not sure, but it looks like we can just do:
```
		clientOptions.Auth.Username = username
		clientOptions.Auth.Password = password
```

Comment (ritbl) [reply agent/utils/mongo_fix/mongo_fix.go:48]: yes, totally, no idea why it wasn't done initially. @artemgavrilov good catch, thank you

**Review by artemgavrilov — APPROVED** [review]

**Comment by JiriCtvrtka** [comment]
There is nil pointer issue for some cases in tests.

**Pull request merged**